### ISSUE TITLE
Does not display change of the blocking state

### ISSUE THREAD

**Issue by rugk** [EFForg/privacybadgerfirefox#730]
See this gif:
![](https://i.imgur.com/TuDro3e.gif)

As you can see I changed this state manually once, but currently PB does not care in any way how I change this.
However when looking at this on the settings page it seems to have changed:
![](https://i.imgur.com/hcOhNU3.png)

So now I don't even know whether it is blocked or not...
BTW when changing the state on the options page it does also not shows the new state...

Privacy Badger 1.0.5
FF 44.0.2

**Comment by cooperq** [comment]
Yep, thats a bug alright. Do you see any errors in the privacy badger console or browser console?

**Comment by rugk** [comment]
> Do you see any errors in the privacy badger console or browser console?

1. There is a PB console??
2. No, there are no errors in the browser console.

**Comment by cooperq** [comment]
There is a PB console indeed! If you [enable addon debugging](https://developer.mozilla.org/en-US/Add-ons/Add-on_Debugger) you can click the debug button for privacy badger in about:addons and get a console. The same debugging messages should show up in the browser console though. 

So, I am curious if this bug is related to issue #711 - are you able to reproduce this in a clean firefox profile with a new install of privacy badger?
Let me know when you have addon debugging set up, and I will ask for some more information.

**Comment by rugk** [comment]
Okay, so I found the culprit: It is again CanvasBlocker.
With it activated the options page reloads when I try to change the sliders.

And yes I've et up addon debugging.

**Comment by cooperq** [comment]
I am also unable to reproduce this issue still even with a fresh profile with just canvas blocker and privacy badger. Are you able to reproduce it with a fresh profile?

**Comment by rugk** [comment]
[githubtest-gif.zip](https://github.com/EFForg/privacybadgerfirefox/files/159813/githubtest-gif.zip)

This is the same gif as in https://github.com/EFForg/privacybadgerfirefox/issues/711, but at the end you can also see this issue happening.
I discovered that it only happens when you use `Ctrl+F` to look (e.g.) for `google`. If you use PrivacyBadgers built-in search function this does not happen.

**Comment by cooperq** [comment]
Can you please give me exact steps to reproduce this on a fresh profile?

**Comment by rugk** [comment]
😟

Okay... once again.

0. Make sure you e.g. have `ssl.google-analytics.com` in your block list.
1. Go to the settings page of PB.
2. Use `Ctrl+F` to search for `google`
3. Click on these sliders.

You'll see that the whole settings page reloads and it does not change the setting (or only does so on the second click).

**Comment by cooperq** [comment]
Thank you!  I can finally reproduce this, and it looks like it is fixed in the current master of privacy badger.
